src/main/java/org/sqlite/SQLiteConnection.java

```diff
@@ -340,9 +369,15 @@ public void setAutoCommit(boolean ac) throws SQLException {
         if (connectionConfig.isAutoCommit() == ac) return;
 
         connectionConfig.setAutoCommit(ac);
-        db.exec(
-                connectionConfig.isAutoCommit() ? "commit;" : connectionConfig.transactionPrefix(),
-                ac);
+        // db.exec(connectionConfig.isAutoCommit() ? "commit;" : this.transactionPrefix(), ac);
+
+        if(this.getConnectionConfig().isAutoCommit()){
+            db.exec("commit;", ac);
+            this.currentTransactionType = null;
+        }else{
+            db.exec(this.transactionPrefix(), ac);
+            this.currentTransactionType = this.getConnectionConfig().getTransactionMode();
```

Comment:
no need to use `this.getConnectionConfig()`, could use `connectionConfig` instead, as it is only set in the constructor and it is marked `final`